### PR TITLE
Social media tags

### LINKED ISSUE #247: Add tags for social media

### PR DESCRIPTION
**Pull request by memolina2323**
## Changes

To include the meta tags for the social media cards, I'm using the Nuxt3 composable [`UseSeoMeta`](https://nuxt.com/docs/api/composables/use-seo-meta).

Previously, in order to include metadata for the head section, the `useHead` composable was being used, but `useSeoMeta` does not only support media cards attributes but the metadata that was handled by `useHead`, so I included that information inside the new composable and removed the previous one.

Open graph (og) cards are the default used ones across the main social media sites. Twitter has their own to implement some specific features but, . So far, there’s no videos or special features for twitter we would like to showcase in our media cards, so I’ll keep it simple by just adding the og ones. This doesn't affect Twitter cards, since it checks for og if no twitter cards attributes are found.

this PR closes #247 

## Implementation details

Here's an example of the attributes added to the Qiskit Metal page:

```
useSeoMeta({
  title: "Qiskit Metal",
  ogTitle: "Qiskit Metal",
  description:
    "A first-of-its-kind, open-source project for engineers and scientists to design superconducting quantum devices with ease: Qiskit Metal. ",
  ogDescription:
    "A first-of-its-kind, open-source project for engineers and scientists to design superconducting quantum devices with ease: Qiskit Metal. ",
  ogImage: "public/images/metal/hero/cryo.png",
  ogUrl: "https://qiskit.org/metal",
});
```
tittle and description are the already implemented head tags.
The image added to most of the pages is the Qiskit logo, but it can be changed. For example in this case, this is the image that's being used:

cryo

### CODE REVIEW

**Review by y4izus — CHANGES_REQUESTED** [review]
Thanks for finding the new composable! In addition to my request, I would suggest to check the recommendations related with the meta descriptions and improve them. You can read more [here](https://developers.google.com/search/docs/appearance/snippet) or [here](https://yoast.com/meta-descriptions/)

File: pages/advocates.vue [hunk 1]
```diff
@@ -20,14 +20,14 @@ definePageMeta({
...
-      content:
-        "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community.",
-    },
-  ],
+  ogTitle: "Qiskit Advocates",
+  description:
+    "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community.",
+  ogDescription:
+    "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community.",
+  ogImage: "public/images/qiskit-logo.png",
```
[the remark this answers is not in the document]
Comment (memolina2323) [reply pages/advocates.vue:30]: good call! sorry, I didn't notice

**Comment by memolina2323** [comment]
Last commit updates descriptions following guidelines from the links provided by Yaiza, including character limitations, using focus keywords, call to action and motivation to visit the page.

**Review by y4izus — DISMISSED** [review]
Awesome! thank you!

File: pages/events/index.vue [hunk 2]
```diff
@@ -130,8 +130,15 @@ definePageMeta({
...
 const upcomingEvents = rawUpcomingEvents as CommunityEvent[];
 
-useHead({
+useSeoMeta({
   title: "Qiskit Events",
+  ogTitle: "Qiskit Events",
+  description:
+    "All past and upcoming events Qiskit related in one place! Hackathons, challenge, camps, talks and more are wainting for you to join!",
+  ogDescription:
+    "All past and upcoming events Qiskit related in one place! Hackathons, challenge, camps, talks and more are wainting for you to join!",
```
Comment (techtolentino) [inline pages/events/index.vue:139]: `wainting` -> `waiting`

File: pages/providers.vue [hunk 3]
```diff
@@ -66,8 +66,15 @@ definePageMeta({
   routeName: "providers",
 });
 
-useHead({
+useSeoMeta({
   title: "Qiskit Providers",
+  ogTitle: "Qiskit Providers",
+  description:
+    "All the ways you can run Qiskit with! from Local Simulators to real Quantum Hardware",
```
Comment (techtolentino) [inline pages/providers.vue:73]: this sentence need to be updated. Maybe something like:

> _All the ways you can run Qiskit! From Local Simulators to real Quantum Hardware_

**Review by techtolentino — CHANGES_REQUESTED** [review]
@memolina2323 - please see my comments... they're minor, but I think it's important we ensure the sentences are correct. 

Thank you! this is a great PR

File: pages/ecosystem.vue [hunk 4]
```diff
@@ -129,15 +129,15 @@ definePageMeta({
...
-  meta: [
-    {
-      name: "description",
-      content:
-        "The Ecosystem consists of projects, tools, utilities, libraries and tutorials from a broad community of developers and researchers. The goal of the Ecosystem is to celebrate, support and accelerate development of quantum technologies using Qiskit.",
-    },
-  ],
+  ogTitle: "Qiskit Ecosystem",
+  description:
+    "The Ecosystem consists of projects, tools, utilities, tutorials... From a broad community of developers and researchers. Learn what can be done using Qiskit!",
```
[the remark this answers is not in the document]
Comment (memolina2323) [reply pages/ecosystem.vue:136]: I had to cut out a few of the examples of things that can go into the ecosystem page to get to the recommended character length... the original paragraph was: "The Ecosystem consists of projects, tools, utilities, libraries and tutorials from a broad community of developers and researchers"

File: pages/events/index.vue [hunk 5]
```diff
@@ -130,8 +130,15 @@ definePageMeta({
 const pastEvents = rawPastEvents as CommunityEvent[];
 const upcomingEvents = rawUpcomingEvents as CommunityEvent[];
 
-useHead({
+useSeoMeta({
   title: "Qiskit Events",
+  ogTitle: "Qiskit Events",
+  description:
+    "All past and upcoming events Qiskit related in one place! Hackathons, challenge, camps, talks and more are wainting for you to join!",
```
[the remark this answers is not in the document]
Comment (memolina2323) [reply pages/events/index.vue:137]: oops, typo

File: pages/providers.vue (hunk 3 above)
Comment (memolina2323) [reply pages/providers.vue:73]: thanks!

File: pages/ecosystem.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (memolina2323) [reply pages/ecosystem.vue]: Due to offline conversations, the description was updated!

**Review by y4izus — DISMISSED** [review]
LGTM!! thank you!

File: pages/ecosystem.vue [hunk 6]
```diff
@@ -129,15 +129,15 @@ definePageMeta({
...
-  meta: [
-    {
-      name: "description",
-      content:
-        "The Ecosystem consists of projects, tools, utilities, libraries and tutorials from a broad community of developers and researchers. The goal of the Ecosystem is to celebrate, support and accelerate development of quantum technologies using Qiskit.",
-    },
-  ],
+  ogTitle: "Qiskit Ecosystem",
+  description:
+    "Discover the Qiskit Ecosystem, a vibrant community accelerating quantum technologies with projects, tools, libraries, and tutorials. Join us now!",
```
Comment (techtolentino) [inline pages/ecosystem.vue:136]: Nice 👏🏼

File: pages/advocates.vue [hunk 7]
```diff
@@ -20,14 +20,14 @@ definePageMeta({
...
-      content:
-        "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community.",
-    },
-  ],
+  ogTitle: "Qiskit Advocates",
+  description:
+    "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community. Join the program today!",
+  ogDescription:
+    "The Qiskit advocate program is a global program that provides support to the individuals who actively contribute to the Qiskit Community. Join the program today!",
+  ogImage: "/images/qiskit-logo.png",
```
Comment (techtolentino) [inline pages/advocates.vue:30]: one thing I forgot: if we want this image to show up in tweet cards, we'll need to use the full URL

`https://qiskit.org/images/qiskit-logo.png`

Using your [preview URL](https://qiskit-org-pr-3300.dcq4xc5i083.us-south.codeengine.appdomain.cloud/events), and a tool like this [Twitter Card Validator](https://tweetpik.com/twitter-card-validator), you'll see that there is no image preview in the twitter card.

Screenshot 2023-06-12 at 6 11 47 AM (2)


However, if you use the full img URL like what was done in the Summer School 2023 page [here](https://github.com/Qiskit/qiskit.org/blob/social-media-tags/pages/events/summer-school-2023.vue#L123-L124), you can see the tweet card produces an image... 

Screenshot 2023-06-12 at 6 12 01 AM (2)

**Review by techtolentino — CHANGES_REQUESTED** [review]
Thanks for addressing my comments @memolina2323 

Just had one last minor request...

**Review by techtolentino — APPROVED** [review]
👏🏼 

great job @memolina2323 

LGTM

**Pull request merged**